source/grid/grid_tools.cc

```diff
@@ -2770,15 +2770,43 @@ namespace GridTools
 
 
 
+  namespace
+  {
+    template <int dim, int spacedim>
+    double diameter(const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+                    const Mapping<dim,spacedim> &mapping)
+    {
+      auto vertices = mapping.get_vertices(cell);
```

Comment:
``const auto &`` ?

Reply:
cannot be a reference. I'll add the const, though. :)